polynote-frontend/polynote/state/error_state.ts

```diff
@@ -78,4 +78,13 @@ export class ErrorStateHandler extends ObjectStateHandler<ErrorState> {
     static notebookRenamed(oldPath: string, newPath: string) {
         ErrorStateHandler.get.update(() => renameKey(oldPath, newPath));
     }
+
+    // if search was able to succeed, should remove ParsingFailures
+    static searchSucceeded() {
```

Comment:
I think it would be better if `ErrorStateHandler` didn't know anything about searching (cf. Separation of Concerns). If this is necessary I'd move it to a search-specific location

Reply:
That makes sense, I was modeling it off of `notebookRenamed`. I'm not sure there's a different way to do this -- the server doesn't send messages that remove errors, and there's no way to access the specific error that should be deleted other than by iterating through and filtering for the `ParsingFailure` string.